src/ccache/util/file.cpp

```diff
@@ -138,16 +153,16 @@ copy_file(const fs::path& src, const fs::path& dest, ViaTmpFile via_tmp_file)
 
   if (fallback_to_rw) {
 #  endif
-    TRY(read_fd(*src_fd, [&](nonstd::span<const uint8_t> data) {
-      write_fd(*dest_fd, data.data(), data.size());
-    }));
+  TRY(read_fd(*src_fd, [&](nonstd::span<const uint8_t> data) {
+    write_fd(*dest_fd, data.data(), data.size());
+  }));
 #  if defined(HAVE_SYS_SENDFILE_H)
-  }
+}
 #  endif
-#endif
 
-  dest_fd.close();
-  src_fd.close();
+dest_fd.close();
+src_fd.close();
```

Comment:
I don't mind the improvement, but I'm hesitant to accept it in its current form since clang-format apparently gets lost in the maze of twisty little ifdefs. Is it possible to restructure to a less twisty form?

Also, note that `copy_file` is only used in [hard link mode](https://ccache.dev/manual/4.9.1.html#config_hard_link) so it's only relevant in a niche use case.